#3360 Should add Well-Understood paper to front page description

daijingz opened the pull request:
In this issue, I believe my overthinking complicated the problem and confused me.
All information is stated in the issue.
#3360 
@JacquesCarette Please take a look.

JacquesCarette reviewed (changes requested):
You've completely misunderstood. The "Well Understood" paper is the new one, the 'old' one is the Drasil position paper.

daijingz commented:
@JacquesCarette Dr.Carette, now I updated both `README.md` and the website. You can see my updates at the end of the `About` section. This time I have uploaded the old position paper, the poster, and the new Well-Understood Paper. (Three things!)
Before some general links had already referred to the old position paper on the webpage, however, they were not targeting to the old position paper directly, so I added a direct link to the old position paper too.

More information will be discussed later tomorrow.

daijingz commented:
@balacij Jason, there is a serious problem with Stable comparisons:
![$S}0B38ED 7NZ%SF}LFVL](https://github.com/JacquesCarette/Drasil/assets/61481741/0fe6504e-6dc1-4484-85ce-eb0c0e6bb64b)
The log temporary files show my changes on the website, which means that the differences were detected with the `stable` repository. However, before I have already tried to use the command `make stabilize` to stabilize everything, but this error still occurs. Because the command `make stabilize` only stabilizes all generated examples, it does not stabilize the website, then I guess this is the problem.

My previous step: `make` -> `make stabilize` -> `make` -> ERROR
The command `make stabilize` does not update my changes on the website copy in the `stable` folder What other commands we can use to update the stable folder?

@JacquesCarette Dr.Carette, up to this step, I am ALMOST DONE with this issue.

JacquesCarette commented on README.md:
```diff
@@ -21,7 +21,7 @@ Generate all the Things! Visit [our website](https://jacquescarette.github.io/Dr
 ## What is Drasil?
 
-For well understood domains, building software ought to be a matter of engineering, based on solid scientific foundations. The ultimate test of "well understood" is being able to teach the domain language to a computer.  Drasil is a framework for generating all of the software artifacts for (well understood) research software, from the natural knowledge base of the domain.
+For well understood domains, building software ought to be a matter of engineering, based on solid scientific foundations. The ultimate test of "well understood" is being able to teach the domain language to a computer.  Drasil is a framework for generating all of the software artifacts for (well understood) research software, from the natural knowledge base of the domain. The new position's Well ...
```
that is very tortured English. Also modern web linking has link text flow in the language itself rather than highlighting that there is a link.

JacquesCarette commented on code/drasil-website/lib/Drasil/Website/About.hs:
```diff
@@ -101,4 +102,13 @@ aboutParagraph8 paperGOOL = S "As described in the" +:+ namedRef paperGOOL (S "G
\ No newline at end of file
+  \and Documents") +:+ S "wiki page"
+
+-- | Paragraph providing a link to the old position's Drasil paper, and a Drasil poster
```
This wording isn't great either.

daijingz commented:
@JacquesCarette @balacij In order to pass the test case, I have to modify the `stable website` with my changes. **Now all test cases passed.**
If you feel my inserted descriptions are not good, I can follow your further guidance. Before I knew the requirements were ambiguous, so my previous strategy was "doing the same thing as our previous action". Some wordings are directly duplicated from `README.md`. -> You know there was no better choice before.

I just read the paper and there is my understanding inside my updates.

While editing styles, I followed the given guidance, where every line cannot be too long.
Any questions please inform me.

daijingz commented:
@JacquesCarette Dr.Carette, from previous feedback, it looks like there are some requirements about writing documentation and websites. Unfortunately, right now my viewable information is too limited.

One important confusion before was the writing styles differences between the Drasil website, README, and the documentations. I feel README is casual because most of its things are flexibly introduced. However, websites and documentations are high-level formal.

This mixture resulted in my messy understanding of writing requirements. I am modifying my writing to formal styles.

daijingz commented:
@JacquesCarette Dr.Carette, I feel this update has already cleanned up all perviously-mentioned problems. Can you have an inspection as soon as possible? I am still checking if there are some extra inconsistencies here.

JacquesCarette commented:
I did check a couple of weeks ago, but it was still full of problems, and I don't have the time to comment on them all - it would take me less time to just fix it all myself. So I did nothing.

balacij commented on README.md:
```diff
@@ -21,7 +21,7 @@ Generate all the Things! Visit [our website](https://jacquescarette.github.io/Dr
 ## What is Drasil?
 
-For well understood domains, building software ought to be a matter of engineering, based on solid scientific foundations. The ultimate test of "well understood" is being able to teach the domain language to a computer.  Drasil is a framework for generating all of the software artifacts for (well understood) research software, from the natural knowledge base of the domain.
+For well understood domains, building software ought to be a matter of engineering, based on solid scientific foundations. The ultimate test of "well understood" is being able to teach the domain language to a computer.  Drasil is a framework for generating all of the software artifacts for (well understood) research software, from the natural knowledge base of the domain. Information of well-und ...
```
It might be nice to instead change the first reference of "well understood" (i.e., the 2nd and 3rd words in this paragraph) into a link `[well-understood](...pdf)`.

balacij replied:
Jingze, what did you think about this suggestion?

balacij replied:
@daijingz

balacij commented on code/drasil-website/lib/Drasil/Website/Body.hs:
```diff
@@ -160,6 +161,15 @@ paperGOOL :: Reference
+posterWiki = makeURI "posterWiki" (danContributionPath
+  ++ "/CAS%20Poster%20Competition/Poster/DrasilPoster.pdf") (shortname' $ S "posterWiki")
+newPaperWiki :: Reference
+newPaperWiki = makeURI "newPaperWiki" (gitHubInfoURL 
```
By "Wiki", we usually either mean wikipedia or our own [wiki](https://github.com/JacquesCarette/Drasil/wiki). I think you can improve the names of these variables: icsePositionPaper, danPoster, and wellUnderstoodPaper are all decent names in line with the others in this file.

balacij commented on code/stable-website/index.html:
```diff
@@ -138,6 +138,12 @@ <h1>About</h1>
         <p class="paragraph">
           A list of papers and documents written about Drasil can be found in the <a href="https://github.com/JacquesCarette/Drasil/wiki/Drasil-Papers-and-Documents">Drasil Papers and Documents</a> wiki page
```
Alternatively, you can also append to this paragraph (fixing its missing period too :smile: ) and inspiration from the `README.md` file, perhaps with something along the lines of:
```
In particular, there is a [position paper](link) outlining our original ideas, a [poster](link), and a [recent paper](link) discussing key concepts.
```

daijingz commented:
@JacquesCarette this is the completion of my second update on this issue. At least all mentioned points have been covered, and I did not see similar problems occurring in my updates. Please have an inspection, and I will be more careful of understanding objects' meanings next time.

One thing you may forget is, that I repaired errors after your suggestions quickly, so that these errors has been recovered within a little time. In order to save your time and get assistance from others, I am going to consult Jason, Sam, and Jiaming.

balacij commented on code/stable-website/index.html:
```diff
@@ -136,7 +136,7 @@ <h1>About</h1>
         </p>
         <p class="paragraph">
-          A list of papers and documents written about Drasil can be found in the <a href="https://github.com/JacquesCarette/Drasil/wiki/Drasil-Papers-and-Documents">Drasil Papers and Documents</a> wiki page
+          A list of papers and documents written about Drasil can be found in the <a href="https://github.com/JacquesCarette/Drasil/wiki/Drasil-Papers-and-Documents">Drasil Papers and Documents</a> wiki page. In particular, there is a  <a href="https://github.com/JacquesCarette/Drasil/blob/master/People/Dan/ICSE%20Workshop%20-%20SE4Science/ICSE_LiterateFrameworkForSCSoftware_LSS.pdf">Old Position ...
```
It looks like there are a few extra spaces in there? @samm82 do you know which Sentence combinators are supposed to be used here?

samm82 commented on code/drasil-website/lib/Drasil/Website/About.hs:
```diff
@@ -99,6 +100,10 @@ aboutParagraph8 paperGOOL = S "As described in the" +:+ namedRef paperGOOL (S "G
 -- | Paragraph providing a link to Drasil papers and documents
-aboutParagraph9 :: Reference -> Sentence
-aboutParagraph9 papersWiki = S "A list of papers and documents written about Drasil can be found in the" +:+ namedRef papersWiki (S "Drasil Papers \
-  \and Documents") +:+ S "wiki page"
```
The reuse of `S "wiki page"` (4 uses in Drasil total) makes me wonder if we want to formally encode it as a named concept (probably in documentation)....

samm82 replied:
@smiths Do you know if we have an issue tracking this sort of thing? I could just make a quick <kbd>[newcomers](https://github.com/JacquesCarette/Drasil/labels/newcomers)</kbd> issue for this one change, but I feel like this is part of a potentially bigger issue (investigating what meaningful concepts are still manually created in examples, etc.)

smiths replied:
@samm82 I'm not aware of an issue for tracking new named concepts.  My guess is that we don't have this, but we might.  We are going to need more newcomer issues when the summer starts, so it is probably best to just create the newcomer issue that you mentioned.  If there are other related isses, we can add the traceability later.

samm82 replied:
Created in #3698!

samm82 commented on code/drasil-website/lib/Drasil/Website/About.hs:
```diff
@@ -99,6 +100,10 @@ aboutParagraph8 paperGOOL = S "As described in the" +:+ namedRef paperGOOL (S "G
+aboutParagraph9 :: Reference -> Reference -> Reference -> Reference -> Sentence
+aboutParagraph9 papersWiki icsePositionPaper danPoster wellUnderstoodPaper = 
+  S "A list of papers and documents written about Drasil can be found in the" +:+ 
+  namedRef papersWiki (S "Drasil Papers and Documents") +:+ S "wiki page. In particular, there is a" +:+ 
+  namedRef icsePositionPaper (S "Old Position Paper") +:+ S "outlining our original ideas, a" +:+ 
+  namedRef danPoster (S "Drasil Poster") +:+ S ", and a" +:+ 
+  namedRef wellUnderstoodPaper (S "Well-Understood Paper") +:+ S "discussing key concepts."
```
I would make the following changes:
1. use "an", since the next word is "old", which starts with a vowel sound
2. remove the repeated articles (i.e., the other "a/an"s) to improve readability (especially for change 4 below)
3. use `+:+.` instead of `+:+` before change 4 below, since the final period "splits off" quite nicely
4. use `foldlList` to format inline lists of items in a consistent way

```suggestion
  namedRef papersWiki (S "Drasil Papers and Documents") +:+ S "wiki page. In particular, there is an" +:+.
  foldlList Comma List [
     namedRef icsePositionPaper (S "Old Position Paper") +:+ S "outlining our original ideas",
     namedRef danPoster (S "Drasil Poster"),
     namedRef wellUnderstoodPaper (S "Well-Understood Paper") +:+ S "discussing key concepts"
  ]
```

Hopefully these changes make sense; I haven't actually run this code, so make sure to test it and make sure that the output is as expected!

samm82 commented on code/drasil-website/lib/Drasil/Website/About.hs:
This change should have caused changes in the generated files; make sure to re-build the artifacts to make sure that they are as expected, and update their stable versions if so!
